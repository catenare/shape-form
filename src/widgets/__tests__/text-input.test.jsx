import React from 'react';
import {fromJS} from 'immutable';
import TextInputWidget from '../text-input';
import ReactShallowRenderer from 'react-test-renderer/shallow';
const renderer = ReactShallowRenderer.createRenderer();

function getProps(props) {
  return {
    schema: fromJS({
      title: 'Test Title'
    }),
    config: fromJS({
      // instructions: 'The quick brown fox jumps over the lazy dog. See [dogs](http://example.com/)'
    }),
    onChange: jest.fn(),
    onBlur: jest.fn(),
    onFocus: jest.fn(),
    value: 'Brown Fox',
    path: 'test',
    ...props
  };
}

describe('TextInputWidget', () => {
  it('renders', () => {
    const props = getProps();

    const tree = renderer.render(<TextInputWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders with instructions', () => {
    const props = getProps({
      config: fromJS({
        instructions: 'The quick brown fox jumps over the lazy dog. See [dogs](http://example.com/)'
      })
    });

    const tree = renderer.render(<TextInputWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
