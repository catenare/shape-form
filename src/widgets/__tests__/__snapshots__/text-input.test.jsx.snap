// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputWidget renders 1`] = `
<ForwardRef(WithStyles)
  autoFocus={false}
  error={false}
  fullWidth={true}
  label="Test Title"
  margin="normal"
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  value="Brown Fox"
/>
`;

exports[`TextInputWidget renders with instructions 1`] = `
<ForwardRef(WithStyles)
  autoFocus={false}
  error={false}
  fullWidth={true}
  helperText="The quick brown fox jumps over the lazy dog. See [dogs](http://example.com/)"
  label="Test Title"
  margin="normal"
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  value="Brown Fox"
/>
`;
